sdk/search/Azure.Search.Documents/src/Batching/SearchIndexingPublisher.cs

```diff
@@ -144,7 +144,7 @@ protected override async Task<bool> OnSubmitBatchAsync(IList<PublisherAction<Ind
                 // Update 'BatchActionCount' so future submissions can avoid this error.
                 BatchActionCount = (int)Math.Floor((double)batch.Count / 2.0);
 
-                AzureSearchDocumentsEventSource.Instance.BatchActionCountUpdated(_sender.Endpoint.AbsoluteUri, oldBatchActionCount, BatchActionCount);
+                AzureSearchDocumentsEventSource.Instance.BatchActionCountUpdated($"{GetType().Name}<{typeof(T).Name}>", _sender.Endpoint.AbsoluteUri, oldBatchActionCount, BatchActionCount);
```

Comment:
I would split this into two events.  Keep BatchActionCountUpdated like you have it here (and can be used in the future when we implement scaling), but we should have something for PayloadTooLarge that stands out as you've clearly mistuned your batch size.